test/tape/crossdb-compatibility.js

```diff
@@ -12,97 +13,99 @@ module.exports = function(knex) {
-      .finally(function() {
+  tape(
+    driverName + ' - crossdb compatibility: setup test table',
+    async function(t) {
+      try {
+        await knex.schema
+          .dropTableIfExists('test_table')
+          .createTable('test_table', function(t) {
+            t.integer('id');
+            t.string('first');
+            t.string('second');
+            t.string('third').unique();
+            t.unique(['first', 'second']);
+          });
+      } finally {
         t.end();
-      });
-  });
+      }
+    }
+  );
 
   tape(
     driverName +
       ' - crossdb compatibility: table may have multiple nulls in unique constrainted column',
-    function(t) {
+    async (t) => {
       t.plan(3);
 
-      knex('test_table')
-        .insert([{ third: 'foo' }, { third: 'foo' }])
-        .catch((err) => {
-          t.assert(true, 'unique constraint prevents adding rows');
-          return knex('test_table').insert([
+      try {
+        await expect(
+          knex('test_table').insert([{ third: 'foo' }, { third: 'foo' }])
+        ).to.be.eventually.rejected;
+        t.assert(true, 'unique constraint prevents adding rows');
+
+        await expect(
+          knex('test_table').insert([
             { first: 'foo2', second: 'bar2' },
             { first: 'foo2', second: 'bar2' },
-          ]);
-        })
-        .catch((err) => {
-          t.assert(true, 'two column unique constraint prevents adding rows');
+          ])
+        ).to.be.eventually.rejected;
 
-          // even one null makes index to not match, thus allows adding the row
-          return knex('test_table').insert([
-            { first: 'fo', second: null, third: null },
-            { first: 'fo', second: null, third: null },
-            { first: null, second: 'fo', third: null },
-            { first: null, second: 'fo', third: null },
-            { first: null, second: null, third: null },
-          ]);
-        })
-        .then(() => {
-          return knex('test_table');
-        })
-        .then((res) => {
-          t.assert(
-            res.length == 5,
-            'multiple rows with nulls could be added despite of unique constraints'
-          );
-        })
-        .finally(() => {
-          t.end();
-        });
+        t.assert(true, 'two column unique constraint prevents adding rows');
+
+        // even one null makes index to not match, thus allows adding the row
+        await knex('test_table').insert([
+          { first: 'fo', second: null, third: null },
+          { first: 'fo', second: null, third: null },
+          { first: null, second: 'fo', third: null },
+          { first: null, second: 'fo', third: null },
+          { first: null, second: null, third: null },
+        ]);
+
+        const res = await knex('test_table');
+        t.assert(
+          res.length === 5,
+          'multiple rows with nulls could be added despite of unique constraints'
+        );
+      } finally {
+        t.end();
+      }
```

Comment:
The logic appears to be equivalent.

(Side note: this really seems like it should be at least 3 separate test cases.  Given the current design of the test case, the test runner will only document assertions that succeeded.  If an assertion fails, the test runner will just report something like:  "Expected 3 assertions  (Only detected 2)" )